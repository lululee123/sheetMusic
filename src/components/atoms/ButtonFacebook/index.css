.button-facebook {
	border-radius: 2px;
	box-shadow: 0 0 8px 0 rgba(50, 52, 50, 0.15);

	&:hover {
		box-shadow: 0 0 4px 0 rgba(50, 52, 50, 0.15);
	}
}
