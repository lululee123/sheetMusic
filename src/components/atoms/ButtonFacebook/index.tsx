import React from 'react';
import classnames from 'classnames';

import FacebookIcon from 'images/icon/facebook.inline.svg';

import ButtonWithIcon from 'components/atoms/ButtonWithIcon';

import styles from './index.css';

interface ButtonFacebookProperty {
	className?: string;
	text?: string;
	onClick: () => void;
}

const ButtonFacebook: React.FC<ButtonFacebookProperty> = ({ className, text, onClick }) => {
	return (
		<ButtonWithIcon
			className={classnames(styles.buttonFacebook, className)}
			Icon={FacebookIcon}
			onClick={onClick}
		>
			{text}
		</ButtonWithIcon>
	);
};

export default ButtonFacebook;
