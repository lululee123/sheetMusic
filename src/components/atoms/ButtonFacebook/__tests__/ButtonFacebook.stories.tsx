import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';

import ButtonFacebook from 'components/atoms/ButtonFacebook';

const stories = storiesOf('atoms/ButtonFacebook', module);

stories.add('__interactive', () => (
	<ButtonFacebook text={text('text', 'Facebook 註冊')} onClick={action('onClick')} />
));
