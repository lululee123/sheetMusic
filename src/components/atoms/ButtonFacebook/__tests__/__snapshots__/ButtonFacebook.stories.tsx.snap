// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <button
    class="button buttonFacebook"
    type="button"
  >
    <test-file-stub />
    <div
      class="main"
    >
      <span>
        Facebook 註冊
      </span>
    </div>
  </button>
</DocumentFragment>
`;
